tensornetwork/__init__.py

```diff
@@ -15,6 +15,7 @@
 from tensornetwork.utils import load_nodes, save_nodes
 from tensornetwork.matrixproductstates.finite_mps import FiniteMPS
 from tensornetwork.matrixproductstates.infinite_mps import InfiniteMPS
+from tensornetwork.backend_contextmanager import DefaultBackend
 
 
 def set_default_backend(backend: Union[Text, BaseBackend]) -> None:
```

Comment:
Add a check that if the default backend stack is empty, and if it's not, raise an `AssertionError`. 

This is to help avoid the confusing situation like: 

```python
with tn.DefaultBackend("tensorflow"):
  tn.set_default_backend("jax")
  a = tn.Node(...)
```
Should `a` be using jax or tensorflow? It's confusing and kinda a footgun, so let's just crash with a clear message instead.